### PR TITLE
env independent tests

### PR DESCRIPTION
**Pull request by osoukup**
- provide the default value for the only var which needs it in the tests and does not have it
- stop syncing the local env variables to the testrunner container to make it local-env-independent
- stop pinning constants which do not need it (mostly because the default value is the same)
- provide one forgotten env var mapping in the osidb-service container compose

### CODE REVIEW

**Review by jobselko — APPROVED** [review]

**Review by JakubFrejlach — CHANGES_REQUESTED** [review]
I still see the env varts in the testrunner container, despite multiple `make stop-local/make start-local` and even `make compose-down`

File: docker-compose.test.yml [hunk 1]
```diff
@@ -36,28 +36,10 @@ services:
...
-        OSIDB_CORS_ALLOW_HEADERS: ${OSIDB_CORS_ALLOW_HEADERS}
-        PRODUCT_DEF_URL: ${PRODUCT_DEF_URL}
-        PRODUCT_DEF_BRANCH: ${PRODUCT_DEF_BRANCH}
-        PS_CONSTANTS_URL: ${PS_CONSTANTS_URL}
-        SNIPPET_CREATION: ${SNIPPET_CREATION}
-        SNIPPET_CREATION_START: ${SNIPPET_CREATION_START}
+      #
+      # Empty environment ensures that all the
+      # tests are independent of the local one
+      environment: []
```
[the remark this answers is not in the document]
Comment (jsvob) [reply docker-compose.test.yml:42]:
```
$ podman exec -it testrunner bash
[root@b59b12fc83c6 src]# declare -p BZIMPORT_BZ_API_KEY
bash: declare: BZIMPORT_BZ_API_KEY: not found
[root@b59b12fc83c6 src]# declare -p BZIMPORT_BZ_URL
bash: declare: BZIMPORT_BZ_URL: not found
[root@b59b12fc83c6 src]# declare -p ET_URL
bash: declare: ET_URL: not found
[root@b59b12fc83c6 src]# declare -p HTTPS_PROXY
bash: declare: HTTPS_PROXY: not found
[root@b59b12fc83c6 src]# declare -p HTTPS_TASKMAN_PROXY
bash: declare: HTTPS_TASKMAN_PROXY: not found
[root@b59b12fc83c6 src]# declare -p JIRA_AUTH_TOKEN
bash: declare: JIRA_AUTH_TOKEN: not found
[root@b59b12fc83c6 src]# declare -p JIRA_TASKMAN_AUTO_SYNC_FLAW
bash: declare: JIRA_TASKMAN_AUTO_SYNC_FLAW: not found
[root@b59b12fc83c6 src]# declare -p JIRA_TASKMAN_PROJECT_KEY
bash: declare: JIRA_TASKMAN_PROJECT_KEY: not found
[root@b59b12fc83c6 src]# declare -p JIRA_TASKMAN_URL
bash: declare: JIRA_TASKMAN_URL: not found
[root@b59b12fc83c6 src]# declare -p JIRA_URL
bash: declare: JIRA_URL: not found
[root@b59b12fc83c6 src]# declare -p OSIDB_CORS_ALLOWED_ORIGINS
bash: declare: OSIDB_CORS_ALLOWED_ORIGINS: not found
[root@b59b12fc83c6 src]# declare -p OSIDB_CORS_ALLOW_HEADERS
bash: declare: OSIDB_CORS_ALLOW_HEADERS: not found
[root@b59b12fc83c6 src]# declare -p PRODUCT_DEF_URL
bash: declare: PRODUCT_DEF_URL: not found
[root@b59b12fc83c6 src]# declare -p PRODUCT_DEF_BRANCH
bash: declare: PRODUCT_DEF_BRANCH: not found
[root@b59b12fc83c6 src]# declare -p PS_CONSTANTS_URL
bash: declare: PS_CONSTANTS_URL: not found
[root@b59b12fc83c6 src]# declare -p SNIPPET_CREATION
bash: declare: SNIPPET_CREATION: not found
[root@b59b12fc83c6 src]# declare -p SNIPPET_CREATION_START
bash: declare: SNIPPET_CREATION_START: not found
[root@b59b12fc83c6 src]# 
```

Did you rebuild the image and recreate the container?

[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply docker-compose.test.yml:42]: I just recreated the container, yes, after rebuilding it works like expected but I wonder why. I thought that these env vars are only sourced upon container running and not embedded on build :thinking:

**Review by jsvob — APPROVED** [review]
Looks like magic to me, but it works. Based on some searching, it seems the variables are also set in the individual tests that need them, so this PR in part removes the duplicates.

File: docker-compose.test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsvob) [reply docker-compose.test.yml:42]: A mystery. IDK.
I found them embedded in metadata both in `.local/share/containers/storage/overlay-images/` and in `.local/share/containers/storage/overlay-containers`. Even though https://docs.docker.com/compose/environment-variables/ says "container" and not "image", https://docs.docker.com/compose/environment-variables/envvars-precedence/ does talk about "Image ENV" and IDK how exactly podman-compose works...

[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply docker-compose.test.yml:42]: whole podman-compose is kinda rigged, when we have some time in the future we should probably migrate to docker-compose.

**Review by JakubFrejlach — APPROVED** [review]

File: docker-compose.test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (osoukup) [reply docker-compose.test.yml:42]: @JakubFrejlach have you solved it at the end or do you still think that this PR does not work? In my case after the image rebuild the env changes are correctly reflected. The environment there looks this way
```
PYTHONUNBUFFERED=1
REQUESTS_CA_BUNDLE=/etc/pki/tls/certs/ca-bundle.crt
LANG=C.utf8
which_declare=declare -f
container=oci
PIP_INDEX_URL=
PWD=/opt/app-root/src
HOME=/root
PIP_NO_CACHE_DIR=off
TERM=xterm
SHLVL=1
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
LESSOPEN=||/usr/bin/lesspipe.sh %s
BASH_FUNC_which%%=() {  ( alias;
 eval ${which_declare} ) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot $@
}
_=/usr/bin/env
```

[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply docker-compose.test.yml:42]: Yeah, after the image rebuild it got solved, I was just confused as I didn't know that these env vars are embeded to the image (and maybe it is a podman-compose bug).
